addons/test_mass_mailing/tests/test_mailing_test.py

```diff
@@ -62,6 +62,34 @@ def test_mailing_test_button(self):
         with self.mock_mail_gateway(), self.assertRaises(Exception):
             mailing_test.send_mail_test()
 
+        # Test if link snippets are correctly converted
+        mailing.write({
+            'subject': 'Subject {{ object.name }}',
+            'preview': 'Preview {{ object.name }}',
+            'body_html':
+                '''<p>
+                Hello
+                    <div class="o_snippet_view_in_browser o_mail_snippet_general pt16 pb16" style="text-align: center; padding-left: 15px; padding-right: 15px;">
+                        <a href="/view">
+                            View Online
+                        </a>
+                    </div>
+                    <div class="o_mail_footer_links">
+                        <a role="button" href="/unsubscribe_from_list" class="btn btn-link">Unsubscribe</a>
+                    </div>
+                </p>'''
+        })
+
+        with self.mock_mail_gateway():
+            mailing_test.send_mail_test()
+
+        record = self.env[mailing.mailing_model_real].search([], limit=1)
+        links = lxml.html.fromstring(self._mails.pop()['body']).xpath('//body/div/a')
+        view_online_button = links[0]
+        unsubscribe_button = links[1]
+        self.assertNotEqual('/view', view_online_button.attrib['href'])
+        self.assertNotEqual('/unsubscribe_from_list', unsubscribe_button.attrib['href'])
```

Comment:
can't we check that '/unsubscribe_from_list' is not in the body ? 
(it will be a little more robust, and we won't need xpath)

same for /view

Reply:
we can check with a regex then :)

(it will still be better than parsing the xml)